tfx/types/artifact_utils.py

```diff
@@ -92,13 +92,20 @@ def get_split_uris(artifact_list: List[Artifact], split: Text) -> List[Text]:
 
   Returns:
     A list of uris of Artifact object in artifact_list with matching split.
+  
+  Raises:
+    ValueError: If number of artifacts matching the split is not equal to 
+      number of input artifacts.
   """
-  matching_artifacts = []
+  result = []
   for artifact in artifact_list:
     split_names = decode_split_names(artifact.split_names)
     if split in split_names:
-      matching_artifacts.append(artifact)
-  return [os.path.join(artifact.uri, split) for artifact in matching_artifacts]
+      result.append(os.path.join(artifact.uri, split))
```

Comment:
Seems we're leaking the details of split uri assembling. Shall we use / enhance existing utility function to handle this? e.g., `artifact_utils.get_split_uris()`.

Reply:
spoke about this at Wednesday sync, marked as resolved.